File: AN_2/SEM_2/MPP/c#/lab6/Model/mpp.model/Utilizator.cs
using MessagePack;
using System;
using System.Security.Cryptography;
using System.Text.Json.Serialization;

namespace Model.mpp.model
{
    [Serializable]
    public class Utilizator : IDentifiable<int>
    {
        public int Id { get; set; }
        public string Nume { get; set; }
        public string Parola { get; set; }

        public Utilizator() {}

        public Utilizator(int id, string nume, string parola)
        {
            Id = id;
            Nume = nume;
            Parola = parola;
        }

        public Utilizator(string nume, string parola)
            : this(0, nume, HashPassword(parola)) {}

        public void SetId(int id) => Id = id;
        public int GetId() => Id;

        public static string HashPassword(string password)
        {
            return BCrypt.Net.BCrypt.HashPassword(password);
        }

        public bool VerifyPassword(string password)
        {
            return BCrypt.Net.BCrypt.Verify(password, Parola);
        }
    }

}

File: AN_2/SEM_2/MPP/java/lab5/Model/src/main/java/ro/mpp2024/model/Utilizator.java
package ro.mpp2024.model;
import org.mindrot.jbcrypt.BCrypt;

import java.io.Serializable;
import java.util.Objects;

public class Utilizator implements Identifiable<Integer>, Serializable {
    private int idUtilizator;
    private String nume;
    private String parola;

    public Utilizator(String nume, String parola) {
        this.nume = nume;
       this.parola = parola;
    }

    public Utilizator(int idUtilizator,String nume, String parola) {
        this.idUtilizator = idUtilizator;
        this.nume = nume;
        this.parola = parola;
    }

    @Override
    public void setId(Integer integer) {
        this.idUtilizator = integer;
    }

    @Override
    public Integer getId() {
        return idUtilizator;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getParola() {
        return parola;
    }

    public void setParolaCriptata(String parola) {
        this.parola = BCrypt.hashpw(parola, BCrypt.gensalt());
    }

    public void setParola(String parola) {
        this.parola = parola;
    }

    public boolean verificaParola(String parola) {
        return BCrypt.checkpw(parola, this.parola);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Utilizator utilizator = (Utilizator) o;
        return idUtilizator == utilizator.idUtilizator;
    }

    @Override
    public int hashCode() {
        return Objects.hash(idUtilizator);
    }

    @Override
    public String toString() {
        return "Utilizator{" +
                "idUtilizator=" + idUtilizator +
                ", nume='" + nume + '\'' +
                ", parola='" + parola + '\'' +
                '}';
    }
}
